index.js

```diff
@@ -501,7 +501,13 @@ class LedgerBridgeKeyring extends EventEmitter {
 
       return undefined
     }
-    window.addEventListener('message', eventListener)
+    window.addEventListener('message', this._eventListener)
+  }
+
+  destroy () {
+    window.removeEventListener('message', this._eventListener)
+    this.currentMessageId = 0
```

Comment:
Nit: Strictly speaking it doesn't seem necessary to restart these properties. Removing the event listener is enough to ensure this instance can be garbage collected, and to ensure no invalid message handling code runs after we've abandoned this instance.